nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -410,3 +412,27 @@ def test__trim_maps():
     np.testing.assert_equal(mask_data, maps_i_mask)
     mask_data[1, 1, 1] = 1  # reset, just in case.
     np.testing.assert_equal(np.asarray(list(range(4))), maps_i_indices)
+
+
+@pytest.mark.parametrize('orig_dtype',
+                         (np.float, np.float32, np.float64, np.int, np.uint),
+                         )
+def test_img_to_signals_labels_int_type(orig_dtype):
+    fake_fmri_data = np.random.RandomState(0).rand(10, 10, 10, 10, ) > 0.5
+    fake_affine = np.random.rand(4, 4)
+    fake_fmri_img_orig = nibabel.Nifti1Image(
+                                        fake_fmri_data.astype(orig_dtype),
+                                        fake_affine,
+                                        )
+    fake_fmri_img_int = new_img_like(fake_fmri_img_orig,
+                                     fake_fmri_data.astype(np.int))
```

Comment:
I think this got mixed up. make `fake_fmri_img_orig` always float, and the other one int, float32, and unsigned int (rename it to something else than `fake_fmri_img_int`